### DISCUSSION TITLE
FlyCap module dll can't be found

### DISCUSSION THREAD

**Discussion by RuyG-O** [bonsai-rx/bonsai#1283] [Q&A]
When trying to run FlyCap we get the following error message: 
![image](https://user-images.githubusercontent.com/126488527/225083384-0b51a810-3b7f-4e33-9e59-0e8f314421a5.png)
We have verified that the module is installed properly. Would you have any suggestions on how to fix this?

**Comment by oliche** [comment]
I have  the slightly similar but different error 

```
A procedure imported by 'FlyCapture2Managed_v100, Version=2.6.3.2, Culture=neutral, PublicKeyToken = 76c6583b4a4585f4' could not be loaded
```

- bonsai 2.6.3
- Flycap 2.12.3.2
- no spinnaker 

I think this needs the spinnaker SDK as mentionned here:
https://groups.google.com/g/bonsai-users/c/4x7EwruXhk8/m/NLysp9_NBAAJ

But who knows what version ? Any recommendation on how to proceed ?

**Comment by bruno-f-cruz** [comment]
Heya. 
If possible, you should use the newer Spinnaker package. As stated in the description, available when you download the package from the package manager, you must install a version-specific SDK. At the moment 1.29.0.5 is compatible with the latest version of the package.

![image](https://user-images.githubusercontent.com/7049351/225128699-f99853d9-5ed0-4662-93b9-b8c700aec984.png)

IF you really must use the flycapture package I believe version `2.13.3.31` of the SDK should be compatible (at least according to the last relevant commit of the package https://github.com/bonsai-rx/pointgrey/commit/8f706237a68f73b9c591d4567cf8516a097ee5bd). You can try to [download it here](https://flir.app.boxcn.net/v/Flycapture2SDK/folder/130120642225)
Hope it helps,
Bruno

**Reply by oliche** [reply]
Ok I'll give it a go today.


What is weird is that I find computers where this package works fine without any Spinnaker SDK, so I suspect the missing library is actually a dependency of the SDK.

I'll update later today, for Bonsai developers I'd suggest to have a more verbose option for this kind of error messages, so we can see what is the library that fails loading !

Thanks for your quick answer !

**Reply by bruno-f-cruz** [reply]
I am not sure I follow. The FlyCapture Package is not dependent on the Spinnaker SDK. So the package should work without it being installed.

**Reply by oliche** [reply]
Yes that what it seems.

So 'FlyCapture2Managed_v100' can't be loaded, and it's unclear what is the missing dependency here .

This is reproducible accross several computers.

Any way to understand where the loading of libraries could fail would help

**Reply by bruno-f-cruz** [reply]
Does this error persist if you install the latest package version and the 2.13.3.31 version of the SDK?

**Reply by oliche** [reply]
![image](https://user-images.githubusercontent.com/15081095/225601663-9e397fa8-4b9c-4bca-8677-7c1de56bb1db.png)
Ok I suppose you're mentionning the flycapture software.

I'll install the x64 Flycapture and update. 

Thanks !

**Reply by oliche** [reply]
So after the package upgrade we get this 
![IMG20230316115158.jpg](https://user-images.githubusercontent.com/15081095/225608683-247f16dc-2f5a-47eb-b754-d8d7ec7194b5.jpg)

And installing the flycap still
 leaves the same DLL loading issue on the original workflow

**Reply by bruno-f-cruz** [reply]
Can you send me the Bonsai.Config file located inside the bonsai install folder? Next to Bonsai.Exe

**Reply by oliche** [reply]
So I've finally ended up with no error by uninstalling everything (SKDs, flycap, Bonsai etc...) and in order:

- install `Bonsai 2.6.3`
- install `FlyCap2Viewer_2.12.3.2_x86.exe`
- install `FlyCap2Viewer_2.12.3.2_x64.exe`
- install Bonsai Packages, in this instance Point Grey `<Package id="Bonsai.PointGrey" version="2.2.0-ivo2" />`

The workflow is finally running !

Thanks for the help and the quick responses.

**Reply by bruno-f-cruz** [reply]
Interesting, not sure what version of the package that one is, but it is not the latest one.
If you want to make sure you are using the latest version, the combination I suggested before should still work!
Otherwise, good luck with experiments!
